### ISSUE TITLE
about Q smoothing loss

### ISSUE THREAD

**Issue by bansh123** [YangRui2015/RORL#1]
https://github.com/YangRui2015/RORL/blob/8924ee56a6df8b5710bbe022494b056243d7a6af/lifelong_rl/trainers/q_learning/sac.py#L250

I think that maybe the Q smoothing loss is implemented differently from the paper.
It seems that the 'diff' is comparing different observations. (instead of comparing original observations and perturbed observations). 

`
diff = noised_qs_pred - qs_pred.repeat(1, 1, size).reshape(self.num_qs,M*size,1)
`
maybe correct?...  or my misunderstanding?

**Comment by YangRui2015** [comment]
Hi, you are right. I have fixed it and reported it in the README.

**Comment by bansh123** [comment]
Thank you for your reply.

However, after fixing, I wonder that whether the performance reported in the paper is still reproducible.

If you check this out later, please let me know too.

-----Original Message-----

**Comment by YangRui2015** [comment]
Sure. In our ablation study, the order of performance contribution is: the OOD loss > the policy smooth loss > the Q smooth loss. And we use a small coefficient 0.0001 for the Q smooth loss. I think the fix can even improve the performance a bit and I will have a check about the difference. 

Thanks a lot！

> Thank you for your reply. However, after fixing, I wonder that whether the performance reported in the paper is still reproducible. If you check this out later, please let me know too.
> […](#)

**Comment by YangRui2015** [comment]
Hi, I have checked several experiments and the results are still within the std range as reported. The benchmark results is shown below. I also noticed that walker2d-medium-v2 task requires slightly larger OOD regularization than before (by tuning the OOD regularization coefficient $\lambda$ and decay pace $d$ ) to stabilize the instability raised from the updated Q smooth loss. I have updated the code with its new hyper parameters.

The benchmark results for hopper and walker2d:
Hopper-random 31.31 +- 0.11 
Hopper-medium 104.60 +- 0.24
Hopper-medium-expert 113.39 +- 1.50
Hopper-medium-reply 102.61 +- 0.44
Hopper-expert 114.43 +- 1.57

Walker2d-random 21.56 +-  0.05 
Walker2d-medium-v2 101.06 +- 2.02
Walker2d-medium-exp 121.24 +- 1.72
Walker2d-medium-replay 93.06 +- 2.99
Walker2d-expert 116.75+-1.43

> Thank you for your reply. However, after fixing, I wonder that whether the performance reported in the paper is still reproducible. If you check this out later, please let me know too.
> […](#)
